packages/horizontal-slider-react/src/HorizontalSlider.tsx

```diff
@@ -0,0 +1,43 @@
+import {
+  createHorizontalSliderComponent,
+  generateHorizontalSliderId,
+  HorizontalSliderProps,
+  RecordWithObjectID,
+  updateNavigationButtonsProps,
+} from '@algolia/ui-components-horizontal-slider-vdom';
+import React, { createElement, Fragment, useEffect, useRef } from 'react';
+
+const UncontrolledHorizontalSlider = createHorizontalSliderComponent({
+  createElement,
+  Fragment,
+});
+
+export function HorizontalSlider<TObject extends RecordWithObjectID>(
+  props: HorizontalSliderProps<TObject>
+) {
+  const listRef = useRef<HTMLOListElement>(null);
+  const nextButtonRef = useRef<HTMLButtonElement>(null);
+  const previousButtonRef = useRef<HTMLButtonElement>(null);
+  const sliderIdRef = useRef(generateHorizontalSliderId());
+
+  useEffect(() => {
+    updateNavigationButtonsProps({ listRef, nextButtonRef, previousButtonRef });
+  });
+
+  return (
+    <UncontrolledHorizontalSlider
+      {...props}
+      listRef={listRef}
+      nextButtonRef={nextButtonRef}
+      previousButtonRef={previousButtonRef}
+      sliderIdRef={sliderIdRef}
```

Comment:
I wonder if this should be a single refs object?

Reply:
probably not due to the diff annoyances that causes

Reply:
I feel like this is easier to understand as single refs, what would your implementation looks like?

Reply:
`refs={{ nextButton: nextButtonRef, previousButton: previousButtonRef, sliderId: sliderIdRef }}` or something similar